### PR TITLE
Propagate parameters on `calcparams_cec` call (#1215)

### LINKED ISSUE #1215: `pvsystem.calcparams_cec()` does not propagate parameters
**Describe the bug**

The function calls `calcparams_desoto` with hardcoded reference values.

https://github.com/pvlib/pvlib-python/blob/40ba4bd5c8b91754aa73e638ed984ab9657847cd/pvlib/pvsystem.py#L1850-L1855

This means the function is silently ignoring its inputs, yielding incorrect results that may go unnoticed.

**Expected behavior**

The function parameters are propagated into the `calcparams_desoto` call. In particular: `EgRef`, `dEgdT`, `irrad_ref`, `temp_ref`

### PR DESCRIPTION
**Pull request by Peque**
- [x] Closes #1215
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [ ] ~~Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.~~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [ ] ~~New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Not sure if tests are required (if so, which tests would be appropriate?). Not sure if an update to "whatsnew" is required.

### CODE REVIEW

**Comment by kanderso-nrel** [comment]
For testing pass-through arguments we typically use the `mocker` fixture with `assert_called_with` like this: https://github.com/pvlib/pvlib-python/blob/master/pvlib/tests/test_modelchain.py#L1633-L1639

The failing tests are unrelated; looks like the PVGIS copyright text just changed again.

**Comment by Peque** [comment]
@kanderso-nrel Thanks!

I also added a comment in https://github.com/pvlib/pvlib-python/issues/776, since I think lint checks could help preventing these types of errors. :blush:

**Review by cwhanse — APPROVED** [review]
Thanks @Peque 

Any changes to the CI tools (linting) should be proposed in a separate PR.

**Comment by wholmgren** [comment]
New/better test would be great but optional since this is straightforward and I'm not too concerned about a regression. Definitely should note it the whats new file.

**Comment by Peque** [comment]
@kanderso-nrel @wholmgren @cwhanse Added test and a short line in whatsnew.

Thanks for your feedback! :blush: :+1:

**Comment by Peque** [comment]
I'll have a look at Python 3.6 and 3.7 failures!

**Comment by Peque** [comment]
Should be fixed now. :blush:

**Review by kanderso-nrel — APPROVED** [review]
Thanks and good catch @Peque!

File: docs/sphinx/source/whatsnew/v0.9.0.rst [hunk 1]
```diff
@@ -164,8 +164,10 @@ Bug fixes
   :py:func:`~pvlib.irradiance.aoi_projection` (:issue:`1185`, :pull:`1191`)
 * Update GFS product names for GFS v16. (:issue:`1202`, :pull:`1203`)
 * Take into account ``EgRef``, ``dEgdT``, ``irrad_ref`` and ``temp_ref`` when
-  calling :py:func:`~pvlib.pvsystem.calcparams_cec`.
-  (:issue:`1215`, :pull:`1216`)
+  calling :py:func:`~pvlib.pvsystem.calcparams_cec` from 
+  :py:meth:`~pvlib.pvsystem.PVSystem.calcparams_cec`. (:issue:`1215`, :pull:`1216`)
```
Comment (Peque) [inline docs/sphinx/source/whatsnew/v0.9.0.rst:168]: It now says:

```rst
[...] when calling from :py:meth:`~pvlib.pvsystem.PVSystem.calcparams_cec`
```

As if it was necessary to call the method to reproduce the issue.

However, this issue affected the function implementation (`pvlib.pvsystem.calcparams_cec`). So calling it not from the method, but directly, was a way to reproduce the issue.

So is this addition required?

Comment (cwhanse) [reply docs/sphinx/source/whatsnew/v0.9.0.rst:168]: I see I have confused things. I'll undo that.

**Review by cwhanse — APPROVED** [review]

**Comment by wholmgren** [comment]
can this be merged?
